src/FluentResults/Results/Result.cs

```diff
@@ -148,6 +148,56 @@ public Result<TNewValue> ToResult<TNewValue>(Func<TValue, TNewValue> valueConver
                 .WithReasons(Reasons);
         }
 
+        /// <summary>
+        /// Convert result with value to result with another value that may fail.
+        /// </summary>
+        /// <example>
+        /// <code>
+        ///  var bakeryDtoResult = result
+        ///     .Bind(GetWhichMayFail)
+        ///     .Bind(ProcessWhichMayFail)
+        ///     .Bind(FormattingWhichMayFail);
+        /// </code>
+        /// </example>
+        /// <param name="valueConverter">Transformation that may fail.</param>
+        public Result<TNewValue> Bind<TNewValue>(Func<TValue, Result<TNewValue>> valueConverter)
```

Comment:
I think it would be good if you can add also a async/await version of it.

Reply:
Last week we added ValueTask support to FluentResults. Please also provide a ValueTask version of the methods.

Reply:
Added overloads